packages/data-context/test/unit/codegen/code-generator.spec.ts

```diff
@@ -1,27 +1,59 @@
-import path from 'path'
-import {
-  codeGenerator,
-  Action,
-  CodeGenResults,
-  CodeGenResult,
-} from '../../src/codegen/code-generator'
-import templates from '../../src/codegen/templates'
+import { parse } from '@babel/parser'
+import { graphqlSchema } from '@packages/graphql/src/schema'
+import { FRONTEND_FRAMEWORKS } from '@packages/types'
 import { expect } from 'chai'
 import dedent from 'dedent'
 import fs from 'fs-extra'
-import { parse } from '@babel/parser'
+import path from 'path'
+import sinon from 'sinon'
+import { DataContext } from '../../../src'
+import { AppApiShape, AuthApiShape, ElectronApiShape, LocalSettingsApiShape, ProjectApiShape } from '../../../src/actions'
+import {
+  Action, codeGenerator, CodeGenResult, CodeGenResults,
+} from '../../../src/codegen/code-generator'
+import { SpecOptions } from '../../../src/codegen/spec-options'
+import templates from '../../../src/codegen/templates'
+import { InjectedConfigApi } from '../../../src/data'
+import { ErrorApiShape } from '../../../src/DataContext'
+import { BrowserApiShape } from '../../../src/sources'
 
 const tmpPath = path.join(__dirname, 'tmp/test-code-gen')
 
+const babelParse = (content: string) => parse(content, { sourceType: 'module', plugins: ['jsx', 'typescript'] })
+
 describe('code-generator', () => {
   before(async () => {
     await fs.remove(tmpPath)
   })
 
+  let ctx: DataContext
+
+  beforeEach(async () => {
+    ctx = new DataContext({
+      schema: graphqlSchema,
```

Comment:
Would it be valuable to have a system test around these, ensuring the specs actually execute? 

I can pull out some of the basic bits from https://github.com/cypress-io/cypress/pull/20164 to help with that if you'd like

Reply:
It would be valuable. What would I need from that PR specifically (what can't I accomplish with the currrent infra) since we already have cypress in cypress?

Reply:
I don't think we'd want to do the full e2e workflow of launchpad -> app, just start w/ a pre-defined system-test project. In that case we should just need the `cy.scaffoldProjectNodeModules(projectName)` command. 

We don't want to run it automatically for every project since we often scaffold projects for the purposes of showing them in global mode and wouldn't want to pay the time of the npm install there.

Reply:
System tests have the ability to install package.json dependencies, so I'm curious why `cy.scaffoldProjectNodeModules` is required in this case. I'm not too familiar with the inner workings of system tests, so just trying to learn more

Reply:
`cy.scaffoldProjectNodeModules` just calls that function used in system tests, it's just that in cases where we scaffold multiple projects to simulate multiple projects in for global mode, or in launchpad where the deps don't matter, we don't want to auto scaffold